suite-native/fiat-rates/src/fiatRatesThunks.ts

```diff
@@ -0,0 +1,201 @@
+import { getUnixTime, subWeeks } from 'date-fns';
+
+import {
+    getFiatRatesForTimestamps,
+    fetchCurrentFiatRates,
+    fetchLastWeekFiatRates,
+} from '@suite-common/fiat-services';
+import { createThunk } from '@suite-common/redux-utils';
+import { FiatCurrencyCode } from '@suite-common/suite-config';
+import { Account } from '@suite-common/wallet-types';
+import { isTestnet } from '@suite-common/wallet-utils';
+import TrezorConnect, { AccountTransaction } from '@trezor/connect';
+
+import { actionPrefix, fiatRatesActions } from './fiatRatesActions';
+import { REFETCH_INTERVAL } from './fiatRatesConst';
+import { selectTickersToBeUpdated, selectTransactionsWithMissingRates } from './fiatRatesSelectors';
+import { TickerId, RateType } from './types';
+
+type UpdateTxsFiatRatesThunkPayload = {
+    account: Account;
+    txs: AccountTransaction[];
+    localCurrency: FiatCurrencyCode;
+};
+
+// This code is not refactored yet, basically it's a copy of the original code from @suite-common
+// TODO: Refactor this to batch requests as much as possible and also fetch rates only for currencies that are actually used
+export const updateTxsFiatRatesThunk = createThunk(
+    `${actionPrefix}/updateTxsRates`,
+    async ({ account, txs, localCurrency }: UpdateTxsFiatRatesThunkPayload, { dispatch }) => {
+        if (txs?.length === 0 || isTestnet(account.symbol)) return;
+
+        const timestamps = txs.map(tx => tx.blockTime!);
```

Comment:
Why did you change this? In suite-common, it's using this function `getBlockbookSafeTime` which returns blockTime safely. Seems weird to access it with `!` right now which could result in crashing the app?

Reply:
`getBlockbookSafeTime` is deprecated and shoudn't be needed anymore. Also blocktime is always present, IDK why it is missing in types.

Reply:
added deprecation message